bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -265,9 +265,14 @@ void ReplicaImp::onMessage<ClientRequestMsg>(ClientRequestMsg *m) {
 
 template <>
 void ReplicaImp::onMessage<ReplicaAsksToLeaveViewMsg>(ReplicaAsksToLeaveViewMsg *m) {
-  LOG_INFO(GL,
-           "Received ReplicaAsksToLeaveViewMsg " << KVLOG(m->viewNumber(), m->senderId(), m->idOfGeneratedReplica()));
-  delete m;
+  if (currentViewIsActive() && m->viewNumber() == getCurrentView()) {
+    LOG_INFO(GL,
+             "Received ReplicaAsksToLeaveViewMsg " << KVLOG(m->viewNumber(), m->senderId(), m->idOfGeneratedReplica()));
+    complainedReplicas.store(std::unique_ptr<ReplicaAsksToLeaveViewMsg>(m));
+    TryToGotoNextView();
+  } else {
+    delete m;
```

Comment:
add WARN log statement to say the `ReplicaAsksToLeaveView` is ignored; also add KVLOG with the relevant values to debug.